drafts/taking-a-byte-of-the-carbon-cake.md

```diff
@@ -0,0 +1,291 @@
+
+With an engineering mindset, it’s tempting to focus on the challenge of improvements to compute workloads; 
+we are trained to optimise and it’s arguably one of the most challenging and fun aspects. 
+Key patterns in this space are 
+demand-shaping (doing work at a rate to match the current carbon-free energy availability in the region), 
+demand-shifting  (moving workloads to a region with lower carbon energy cost) and 
+time-shifting (deferring until a time when carbon-free energy is available). There’s already some useful tools in this space, 
+such as a Kubernetes (carbon-aware) scheduler.
+
+However with the trend in cloud providers towards fully-green energy, it’s likely that their data centres in the most popular regions will be 100% soon, and so the gains in moving workloads from one to another are decreasing. This diminishes the benefits of demand shifting and demand shaping, leaving time-shifting as a reasonable option.
+
+Therefore we purposefully ignore demand-shaping and demand-shifting in the following opinionated view of the activities with the most significant impact; 
+focusing instead on three groups; easy-wins, medium effort and those activities at architecture design time:
+
+
+### First: Low-Hanging Fruit
+
+![Low hanging fruit](https://cdn.hashnode.com/res/hashnode/image/upload/v1730804092300/ebrrwNFL7.jpeg?auto=format)
+
+* **Measure**
+
+  Start with carbon footprint tools, which will provide a ball-park figure of carbon usage decomposed over services. Be wary that these may not include ‘free usage’ (where free means cost-free of course) and may lack accuracy on shared ‘serverless’ services, eg: lambda functions.  Third party data from CCF or ClimateIQ data provides more granular detail.
+
+
+* **Switch off**
+
+  Particularly relevant for dev and test envs, but also look for resources or services no longer actively used, for example S3 buckets, old logs. 
+  See [GSF pattern: delete unused storage](https://patterns.greensoftware.foundation/catalog/cloud/delete-unused-storage-resources)
+
+
+* **Scale appropriately**
+
+  Ramp down for quiet business periods.  Ramp down environments or services as appropriate.  Consider containerising or co-hosting services where CPU usage is low, such as 
+  [GSF pattern: optimize CPU utilization](https://patterns.greensoftware.foundation/catalog/cloud/optimize-avg-cpu-utilization).
+
+### Second: Medium Effort Activities
+
+* **Review Failover**
+
+  Re-examine fail-over strategy to reduce under utilised resources; 
+  ie: does your system really need active-active? [AWS Well Architected: Planning for Disaster](https://docs.aws.amazon.com/wellarchitected/2022-03-31/framework/rel_planning_for_recovery_disaster_recovery.html) .  
+  This may allow some to be switched off, or moved to colder options with longer failover times.
```

Comment:
would infrastructure-as-code be worth mentioning here as a slower alternative to hot-standby